### ISSUE TITLE
Age correction across the life course

### ISSUE THREAD

**Issue by eatyourpeas** [rcpch/digital-growth-charts-server#127]
At project board today it was decided that age correction until 2 y in under 32 week gestation, and until 1 y corrected age for <37 weeks had been created for the paper charts but could now be abandoned as the api can accurately correct across the life course. This might create a crowded chart and so it was suggested a toggle button be applied to the chart to show/hide the chronological age.

**Comment by eatyourpeas** [comment]
@statist this is my starter for 10

```
def corrected_decimal_age(birth_date: date, observation_date: date, gestation_weeks: int, gestation_days: int)->float: 
    """
    Corrects for gestational age. Does not account for term.
    :param birth_date: date of birth
    :param observation_date: date observation made
    :param gestation_weeks: weeks of gestation up to 42
    :param gestation_days: days in excess of weeks
    :chronological_decimal_age is a dependency
    """
    
    correction_days = 0

    if gestation_weeks == 0:
        gestation_weeks=40
    
    pregnancy_length_days = (gestation_weeks * 7) + gestation_days

    ## age correction
    correction_days = TERM_PREGNANCY_LENGTH_DAYS - pregnancy_length_days
    edd = birth_date + timedelta(days=correction_days)
    corrected_age =  chronological_decimal_age(edd, observation_date)

    return corrected_age
```

**Comment by eatyourpeas** [comment]
By the way, this only tangentially relevant. The concept of term itself is called into question in [bmjopen](https://bmjopen.bmj.com/content/9/4/e025275): cognitive scores are significantly different between babies born early term and 40 weeks. Perhaps this is just an association so caution need, but by correcting for all ages of gestation it may be easier to decide if this is genuine, and if so not reinforce the idea of 37 weeks being as good as 40 weeks, if that turns out not to be the case.

**Comment by BB-1313** [comment]
Hi all,
Have you as the development group or the board had considered who the first "clients" might be?
Given they have achieved I believe 100% coverage in terms of UK neonatal units is it worth engaging directly with CleverMed (Badgernet) regarding the API at an early stage?
Basil

**Comment by pacharanero** [comment]
BadgerNet came to one of our Webinars and are interested already yes. John Badger said in the comments "this is manna from heaven".

**Comment by BB-1313** [comment]
Good to know

I wonder why they hadn’t considered doing it internally for a while but that seems the way in the commercial/proprietary world I suppose?!

**Comment by pacharanero** [comment]
Age correction across all lie course (not stopping at 2 yrs or 1 yr) is now enabled and is in testing

**Comment by statist7** [comment]
@eatyourpeas this afternoon seemed to be rowing back from the idea of providing a toggle to switch between unadjusted and adjusted age on the chart, on the grounds that it's not obvious which should be the default.

A solution would be to use the status quo, i.e. to default to
- unadjusted age for term babies
- adjusted age up to 1 year for preterm babies, and 
- adjusted age up to 2 years for very preterm babies.

**Comment by eatyourpeas** [comment]
Thank you @statist7 . This is a charting issue so will move it to the charting package. I am very happy to implement it - I would love to show you how it works at the moment before making the change though.

**Comment by statist7** [comment]
I'd love to see it. Do send me a zoom invite - tomorrow is free for me.

**Comment by eatyourpeas** [comment]
This was closed with PR af456e2
